slides/linear_model.md

```diff
@@ -0,0 +1,245 @@
+
+???
+An error for each point correspond to the red bar in the figure
+the best fit is the line which minimize the sum of (the square of) those red lines.
+
+Fortunatly, there is a formula, given **X** and **y**, to find the optimal weights in an efficient manner
+
+---
+# Linear regerssion in higher dimension
+
+.pull-left.shift-left[<img src="../figures/lin_reg_3D.svg" width="130%">]
+
+
+???
+In real world data set, **X** has severals dimensions, and it is not possible anymore to represent it.
+
+---
+# Logistic regression
+
+For **classification** we use a logistic regression model  
+**y** is binary, either +1 or -1
+
+.shift-left.pull-left[<img src="../figures/categorical.svg" width="100%">]
+.pull-right.very-small[
+```python
+From sklearn.linear_model import LogisticRegression
+log_reg = LogisticRegression()
+```
+]
+???
+Logistic regression is a linear model for **classification** - not regresion as the name suggest.
+
+In our adult_census dataset, we do not have continous value for salary but only whether the salary is higher than $50K or not.
+
+
+---
+# Logistic regression
+
+The output is now conposed with the logistic function
+Logistic(x) = 1 / (1 + exp(-x))
+
+.shift-left.pull-left[<img src="../figures/logistic_color.svg" width="100%">]
+.pull-right.very-small[
+```python
+From sklearn.linear_model import LogisticRegression
+log_reg = LogisticRegression()
+log_reg.fit(X, y)
+```]
+
+---
+# Logistic regression in 2 dimensions
+
+**X** is 2-dimensional  
+**y** is the color
+
+.shift-left.pull-left[<img src="../figures/logistic_2D.svg" width="100%">]
+.pull-right[<img src="../figures/logistic_3D.svg" width="100%">]
+
+???
+Here is an other way of representing our data.
+In this case, X has two dimension x1 and x2.
+The axis correspond to x1, x2
+and the color correspond to the target label y 
+
+---
+# Model complexity
+Salary = *.4 x* education_level + *.2 x* hours-per-week + *.1 x* age + 
+.red[ *.2 x* zodiac_sign + *.1 x* red_socks] + ...  
+
+* Linear model could also overfit. 
+ - Reducing its complexity by reducing its parameters value is called *Ridge regression*
+???
+
+If we have too many parameters w.r.t. number of samples, it's adviced to penalize the parameters of our models.
+
+With weights penalty, we include the value of the model's weights within the objectif function
+So a penalized model should choose lower weights for almost a similar fit
+
+The complexity parameter \alpha controls the amount of shrinkage: the larger the value of \alpha, the greater the amount of shrinkage and thus the coefficients become more robust to collinearity.
+
+---
+# Bias-variance tradeoff: an example
+
+
+.pull-left.shift-left[<img src="../figures/lin_reg_2_points.svg" width="110%">]
+
+---
+# 2 points example
+
+
+.pull-left.shift-left[<img src="../figures/lin_reg_2_points_no_penalty.svg" width="110%">]
+.pull-right[<img src="../figures/lin_reg_2_points_ridge.svg" width="110%">]
```

Comment:
Can you explore adding test data points in light gray that are drawn from the original model with a bit of noise. The goal would be to show that a bit of bias can minimize the left-out error.

These points could appear in a third slide: keeping the two first figures a such, and adding a slide after.

Reply:
I added some grey points (cf slide_linear.html).
But here the regularization do not help those points, i.e. it do not minimize the left-out error.

Nonetheless, I would be happy having a visual where bias could help!

Reply:
I think that we will need some legend to know that the grey points are the testing points.

Shall we have the fitted line in the same colours than the 2 points that they fit.
I am also wondering which palette of colour to use for colorblindness here. We could play with marker types maybe to differentiate?

Reply:
I had remove the grey point (still visible in "presentation mode")
It do not allow to see that regularization could be helpful.